spec/index.bs

```diff
@@ -5131,70 +5144,114 @@ enum GPUStoreOp {
 
         <div algorithm="GPURenderEncoderBase.setPipeline">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The render pipeline to use for subsequent drawing commands.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |pipeline| is [=valid=].
+                        - |pipeline|.{{GPUObjectBase/[[device]]}} is |this|.
+                    </div>
+                1. Set |this|.{{GPURenderEncoderBase/[[pipeline]]}} to be |pipeline|.
+            </div>
         </div>
 
     : <dfn>setIndexBuffer(buffer, indexFormat, offset, size)</dfn>
     ::
+        Sets the current index buffer.
 
         <div algorithm="GPURenderEncoderBase.setIndexBuffer">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setIndexBuffer(buffer, indexFormat, offset, size)">
-                buffer:
-                indexFormat:
-                offset:
-                size:
+                |buffer|: Buffer containing index data to use for subsequent drawing commands.
+                indexFormat: Format of the index data contained in |buffer|.
+                |offset|: Offset in bytes into |buffer| where the index data begins.
+                |size|: Size in bytes of the index data in |buffer|.
+                    If `0`, |buffer|.{{GPUBuffer/[[size]]}} - |offset| is used.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setIndexBuffer()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |buffer| is [=valid=].
+                        - |buffer|.{{GPUObjectBase/[[device]]}} is |this|.
+                        - |offset| &le; |buffer|.{{GPUBuffer/[[size]]}}.
+                        - |size| &le; |buffer|.{{GPUBuffer/[[size]]}} - |offset|.
+                    </div>
+                1. Add |buffer| to the [=usage scope=] as
+                    {{GPUBufferUsage/INDEX|GPUBufferUsage.INDEX}}).
+                1. Set |this|.{{GPURenderEncoderBase/[[index_buffer]]}} to be |buffer|.
```

Comment:
Will soon need the index format in the current state, for validation. The rest of the args aren't needed for validation, but will eventually be needed when we want to describe the actual behavior of the draw call (someday). Optionally could track them now, same with setVertexBuffer.